### PR TITLE
Modified pseudo GT creation.

### PR DESCRIPTION
**Pull request by sriamazingram**
Other changes:

- Added CPUs option to context match. 
- Added option to external embedding. (pseudo-gt)

### CODE REVIEW

File: tl/cli/context-match.py [hunk 1]
```diff
@@ -30,9 +32,14 @@ def add_arguments(parser):
                         help='The minimum threshold for similarity with input context for quantity matching. '
                              'Default: 0.85')
     parser.add_argument('--custom-context-file', type=str, dest='custom_context_file', required=False,
-                        help="The file is used to look up context values for matching.")
-    parser.add_argument('--string-separator', action='store', type=str, dest='string_separator', default=",",
-                        help="Any separators to separate from in the context substrings.")
+                        help="The file is used to look up context values for matching.") 
+    parser.add_argument('--string-separator', action = 'store', type=str, dest = 'string_separator', default = ",", 
+                        help = "Any separators to separate from in the context substrings.")
+    parser.add_argument('--use-cpus', action='store', type=int,
```
Comment (saggu) [inline tl/cli/context-match.py:38]: what is the default value? Any time required=False is specified, it should have a reasonable default value

File: tl/cli/create-pseudo-gt.py [hunk 2]
```diff
@@ -29,6 +29,11 @@ def add_arguments(parser):
                              "with corresponding thresholds; column:threshold"
                              "; multiple features can be specified "
                              "by separating with a comma.")
+    
+    parser.add_argument('--filter', type=str, action='store',
```
Comment (saggu) [inline tl/cli/create-pseudo-gt.py:33]: add example in the help string

File: tl/features/create_pseudo_gt.py [hunk 3]
```diff
@@ -2,23 +2,80 @@
...
+        col, val = filter.split(":")
+        val =  float(val)
+        if col not in df.columns:
+            raise RequiredColumnMissingException("The input column {} does not" 
+                                                 " exist in given data.".format(col))
     ffv = FFV()
     if ffv.is_candidates_file(df):
+        grouped = df.groupby(by=["column", "row"])
         for column, threshold in column_thresholds:
             if column not in df.columns:
```
Comment (saggu) [inline tl/features/create_pseudo_gt.py:23]: this is already checked at line 16, no need to check again. Also what the difference between column threshold and filter?

File: tl/features/create_pseudo_gt.py [hunk 4]
```diff
@@ -2,23 +2,80 @@
...
+            top_cd_df = pd.concat([gdf.sort_values(by=[column], ascending=False).head(1) for _, gdf in grouped])
+            if filter:
+                top_cd_df = top_cd_df[top_cd_df[col] > val]
+            if threshold=="median":
+                for _, gdf in top_cd_df.groupby(by=["column"]):
+                    gdf.loc[(gdf[column].astype(float) >= gdf[column].astype(float).median()), 
+                            output_column] = 1
+                    df.loc[gdf.index[gdf[output_column].astype(float) == 1],
+                           output_column] = 1
+            elif threshold=="mean":
```
Comment (saggu) [inline tl/features/create_pseudo_gt.py:37]: PEP8 incompatible

**Review by saggu — CHANGES_REQUESTED** [review]
minor changes/clarifications required

File: tl/features/create_pseudo_gt.py (hunk 3 above)
Comment (sriamazingram) [reply tl/features/create_pseudo_gt.py:23]: In line 16, the column in filter is checked. Filter argument is to filter out the rows based on values. Columns in column threshold is used to make the pseudo gt.

**Pull request merged**